### ISSUE TITLE
Excluding unwanted changes while saving as TMDL from analysis server- "state" column 

### ISSUE THREAD

**Issue by rohitghegde** [TabularEditor/TabularEditor#1222]
**Describe the bug**
I am trying to implement a devops process using TMDL model for PowerBI datasets, and when I save the tmdl model into my local I am gettting the unwanted "state" column in all the files. This creates problem while creating Pull request to the github main branch, which ends up wih too many file changes. 

I am also checking "Ignore inferred properties" in Preferences as below:
![image](https://github.com/user-attachments/assets/f5e1e4a8-ec30-4fd6-926e-5cd284c11b08)

**To Reproduce**
1. Deploy the tmdl model from github main branch into Power BI dataset - workspace 'A'. 
2. Take the copy of it into another PowerBI workspace 'B' for development purpose. 
3. After all the development changes, save the dataset as tmdl from workspace 'B' into a github feature branch 'feature_b'
4. Create pull request from 'feature_b' to github main. 

**Expected behavior**
The **State** column must be excluded while saving the semantics model as TMDL into local

**Observed behavior**
Too many files with "state: calculationNeeded"
![image](https://github.com/user-attachments/assets/b37f2521-8e4b-4ff3-8b57-bcadffb16b70)

**Application specifics**
 - Tabular Editor 2.24.1
 - Semantic Engine used:  Azure Analysis Server

Would also be interesting if there is a way to do this in command line.

**Comment by otykier** [comment]
I'm afraid the "Ignore inferred properties" option is not used by the TMDL serializer.

I see you're using TE 2.24.1. Have you tried using the latest version [2.25](https://github.com/TabularEditor/TabularEditor/releases/tag/2.25)? I'm asking because the latest version uses a newer version of AMO/TOM, where the TMDL serializer may have been improved.

Tagging @mthierba @ruiromano for visibility.

**Comment by mthierba** [comment]
That's odd - those properties should never be part of a TMDL output. They certainly weren't in earlier releases of TMDL, I wonder if that's a regression? Will raise with the team internally.

**Comment by rohitghegde** [comment]
Thank you! @otykier @mthierba 

I will give it a try with version 2.25

So the only solution is to wait until next version ? or 
could be I have my own compiled version ? if I am able to identify the issue. For this i would like to clarify regarding the license. 
I would like to use it personally and not re-distribute it.

**Comment by mthierba** [comment]
@rohitghegde If you wanted to address this issue immediately, `pbi-tools` can remove selected properties for you via a declarative settings file.

With this `.pbixproj.json` in the working directory (and the `pbi-tools.exe` on your PATH):
```json
{
    "settings": {
        "model": {
            "ignoreProperties": [
                "state"
            ]
        }
    }
}
```

...you can run:

    pbi-tools convert {your-tmdl-folder-path} -settingsFile .\.pbixproj.json  -overwrite

That will read your TMDL project from the specified folder and serialize it back to the same location, dropping all "state" properties along the way.

You can also serialize into a separate output folder like so:

    pbi-tools convert {your-tmdl-folder-path} {your-output-path} -settingsFile .\.pbixproj.json  -overwrite

Obtain the tool here: https://github.com/pbi-tools/pbi-tools/releases

**Comment by otykier** [comment]
> So the only solution is to wait until next version ?
> could be I have my own compiled version ? if I am able to identify the issue.

If the issue isn't solved in 2.25, then I'm afraid there's not much we can do, until Microsoft provides updates TOM client libraries where the issue is fixed.

> For this i would like to clarify regarding the license. I would like to use it personally and not re-distribute it.

It's MIT license, meaning you can do whatever you want with it personally.

**Comment by rohitghegde** [comment]
Hi @otykier @mthierba 

Seems I was using 2.24 version before, and I tried to save as TMDL model with both 2.25 and 2.24.1 version via command line as below: 

`TabularEditor.exe "Data Source=%_datasource%;User ID=app:${{ secrets.AZURE_SPN_USER }}@${{ secrets.AZURE_SPN_TENANT }};Password=${{ secrets.AZURE_SPN_SECRET }}" "%pbi-dsid%" -TMDL "../${{ inputs.target-folder }}" -G`

**Interestingly, 2.24.1 works as expected, it excludes state column 
But both 2.24 and 2.25 version includes state column while saving the model.** 

I am able to replicate this only via command line and not via GUI

**Comment by RuiRomano** [comment]
We fixed the serialization of inferred properties back in April. In TMDL, inferred properties should never be serialized.

I'm unable to repro your behavior using TE 2.25, by connecting to a model with an unprocessed calculation column.

**Comment by rohitghegde** [comment]
@RuiRomano Thank you for trying it out, as I said I am only able to replicate this only via command line and not via GUI. 

Could you please check it via command line ? 

**Interestingly, 2.24.1 works as expected, the state is not serialized.
But both 2.24 and 2.25 version includes state column while saving the model.**

**Comment by RuiRomano** [comment]
@otykier can you get a repro of this? Its strange that only replicates using the command line and not GUI.
I talked with TMDL engineers and this was definitely fixed back in April.

**Comment by otykier** [comment]
@RuiRomano I tried with the latest TE version (v. 2.25 which uses AMO/TOM v. 19.84.1), but I am not able to reproduce this behavior, neither via command line nor GUI. It would also surprise me if command line vs. GUI produced different results, since it's the same code path that ultimately produces the TMDL.

@rohitghegde are you sure that you used the latest version (2.25) for CLI execution in your DevOps pipeline? The output log from your CLI pipeline should indicate which build # of Tabular Editor is being used.

**Comment by rohitghegde** [comment]
Hi @otykier @RuiRomano  Yes you are right, I apologize . I got confused with the 2.24.0 build. 

I did another test and seems this issue is fixed from version 2.24.1 onwards and is also working fine with 2.25.0.
Thanks alot once again for suggesting to use 2.25.0
